src/pages/challenge/component.js

```diff
@@ -0,0 +1,118 @@
+import React, {useEffect} from "react";
+import {useHistory, useParams} from "react-router-dom";
+import Header from "../../components/header";
+import Footer from "../../components/footer";
+import Challenge from "../../components/challenge";
+import {Button} from "antd";
+import {Link} from "react-router-dom";
+import styles from "./_.module.css";
+
+import cameraIcon from "../../images/video-camera-white.svg";
+import crossIcon from "../../images/cross.svg";
+import checkBlackIcon from "../../images/check-black.svg";
+import checkWhiteIcon from "../../images/check-white.svg";
+import backIcon from "../../images/chevron-left.svg";
+import userImg from "../../images/user.png";
+
+const BackIcon = ({className, onClick}) => (
+  <button className={styles.goBack} onClick={onClick}>
+    <img src={backIcon} alt="Go back" className={className} />
+  </button>
+);
+
+const UserImage = ({userImg, className}) => (
+  <img src={userImg} alt="User" className={className} />
+);
+
+export const ChallengePage = ({fetchChallenge, challenge, engageChallenge, user}) => {
+  const history = useHistory();
+  let {id} = useParams();
+  const challengeId = decodeURIComponent(id);
+  const username =
+    challenge &&
+    challenge.challengeState.challengeDefinition.challengeOwnerHandle.assetId.id;
```

Comment:
it is not clear for me, 
```challenge.challengeState.challengeDefinition.challengeOwnerHandle.assetId.id``` - here you try to get ID, but your variable is called as username